src/gui/editors/TimeLineWidget.cpp

```diff
@@ -92,6 +98,7 @@ void TimeLineWidget::addToolButtons( QToolBar * _tool_bar )
 	autoScroll->addState(embed::getIconPixmap("autoscroll_stepped_on"), tr("Stepped auto scrolling"));
 	autoScroll->addState(embed::getIconPixmap("autoscroll_continuous_on"), tr("Continuous auto scrolling"));
 	autoScroll->addState(embed::getIconPixmap("autoscroll_off"), tr("Auto scrolling disabled"));
+	autoScroll->changeState(static_cast<int>(m_autoScroll));
```

Comment:
Yeah, this is odd. Why store the state separately in `TimeLineWidget` when that state is already in the `NStateButton`?

Reply:
> Hmm, another thing is that the NStateButton is stored on the stack in addToolButtons(), so it can't be accessed outside of that function. Unless we want to make it a member variable?

Not stored on the stack.. its stored on the heap, but scoped within the `addToolButtons` function.

But also.. I just realized that `addToolButtons` is for some reason in this class. I think it should be in the `Editor` class (or even just directly done within that constructor??) Maybe that is out of scope though.